logprep/processor/base/exceptions.py

```diff
@@ -70,20 +70,28 @@ def __init__(self, message: str, rule: "Rule"):
 class ProcessingWarning(Warning):
     """A warning occurred - log the warning, but continue processing the event."""
 
-    def __init__(self, message: str, rule: "Rule", event: dict, tags: List[str] = None):
+    def __init__(self, message: str, event: dict, rule: "Rule" = None, tags: List[str] = None):
         self.tags = tags if tags else []
-        rule.metrics.number_of_warnings += 1
-        message = f"{message}, {rule.id=}, {rule.description=}, {event=}"
+        if rule:
+            rule.metrics.number_of_warnings += 1
+            message += f", {rule.id=}, {rule.description=}"
+        message += f", {event=}"
```

Comment:
I ensured the visibility again by adding the rule to the `add_fields_to` method. I also reverted the exception signature to the original, but I had to leave this `if rule`, because fields that are added outside of processors don't have rules, and they can still raise a `FieldExistsWarning`.